File: Excel/424 Challenge.py
import pandas as pd

input = pd.read_excel('424 Insert in Between Multiplication.xlsx', usecols='A', nrows=9)
test  = pd.read_excel('424 Insert in Between Multiplication.xlsx', usecols='B', nrows=9)

def transform_number(number):
    digits = [int(digit) for digit in str(number)]
    transformed_digits = [digits[i] * digits[i+1] for i in range(len(digits)-1)]
    transformed_digits.append('')
    mixed_digits = [digit for pair in zip(digits, transformed_digits) for digit in pair]
    result = ''.join(map(str, mixed_digits))
    return result

input["Answer Expected"] = input["Words"].apply(transform_number)

print(input["Answer Expected"].equals(test["Answer Expected"])) # True

File: Excel/400-499/424/424 Challenge.R
library(tidyverse)
library(readxl)

input = read_excel("Excel/424 Insert In Between Multiplication.xlsx", range = "A1:A10")
test  = read_excel("Excel/424 Insert In Between Multiplication.xlsx", range = "B1:B10")

transform_number = function(number){
  str_number = as.character(number)
  digits = strsplit(str_number, "")[[1]] %>% as.numeric()
  ndigits = length(digits)
  products = map(1:(ndigits - 1), ~{
    digits[.x] * digits[.x+1]
  }) %>% c(., "")
  result = map2(digits,products,  ~{
    c(.x, .y)
  }) %>% unlist() %>%
    paste(collapse = "")
  return(result)
}

result = input %>%
  mutate(`Answer Expected` = map_chr(Words, transform_number)) %>%
  select(-Words)

identical(result, test)
# [1] TRUE
